core/src/main/java/brooklyn/BrooklynVersion.java

```diff
@@ -46,201 +58,333 @@
+        private final String version;
+        private final String lastModified;
+        private final Map<String, String> additionalData;
+
+        BrooklynFeature(String name, String symbolicName, String version, String lastModified, Map<String, String> additionalData) {
+            this.symbolicName = checkNotNull(symbolicName, "symbolicName");
+            this.name = name;
+            this.version = version;
+            this.lastModified = lastModified;
+            this.additionalData = ImmutableMap.copyOf(additionalData);
+        }
+
+        /** @return Present if any attribute name begins with {@link #BROOKLYN_FEATURE_PREFIX}, absent otherwise. */
+        private static Optional<BrooklynFeature> newFeature(Attributes attributes) {
+            Map<String, String> additionalData = Maps.newHashMap();
+            for (Object key : attributes.keySet()) {
+                if (key instanceof Attributes.Name && key.toString().startsWith(BROOKLYN_FEATURE_PREFIX)) {
+                    Attributes.Name name = Attributes.Name.class.cast(key);
+                    String value = attributes.getValue(name);
+                    if (!Strings.isBlank(value)) {
+                        additionalData.put(name.toString(), value);
+                    }
+                }
+            }
+            if (additionalData.isEmpty()) {
+                return Optional.absent();
+            }
+
+            // Name is special cased as it a useful way to indicate a feature without
+            String nameKey = BROOKLYN_FEATURE_PREFIX + "Name";
+            String name = Optional.fromNullable(additionalData.remove(nameKey))
+                    .or(Optional.fromNullable(Constants.BUNDLE_NAME))
+                    .or(attributes.getValue(Constants.BUNDLE_SYMBOLICNAME));
+
+            return Optional.of(new BrooklynFeature(
+                    name,
+                    attributes.getValue(Constants.BUNDLE_SYMBOLICNAME),
+                    attributes.getValue(Constants.BUNDLE_VERSION),
+                    attributes.getValue("Bnd-LastModified"),
+                    additionalData));
+        }
+
+        public String getLastModified() {
+            return lastModified;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public String getSymbolicName() {
+            return symbolicName;
+        }
+
+        public String getVersion() {
+            return version;
+        }
+
+        /** @return an unmodifiable map */
+        public Map<String, String> getAdditionalData() {
+            return additionalData;
+        }
+
+        @Override
+        public String toString() {
+            return getClass().getSimpleName() + "{" + symbolicName + (version != null ? ":" + version : "") + "}";
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(symbolicName, version);
+        }
+
+        @Override
+        public boolean equals(Object other) {
```

Comment:
Brooklyn supports loading different bundles with the same symbolicName+version, namespaced in different catalog items. Might not be worth the trouble supporting them here, but definitely it's something to have in mind.